fem/gridfunc.hpp

```diff
@@ -154,16 +154,35 @@ class GridFunction : public Vector
 
    int GetFaceValues(int i, int side, const IntegrationRule &ir, Vector &vals,
                      DenseMatrix &tr, int vdim = 1) const;
+   /*
+    void GetVectorValues(ElementTransformation &T, const IntegrationRule &ir,
+                         DenseMatrix &vals) const;
 
-   void GetVectorValues(ElementTransformation &T, const IntegrationRule &ir,
-                        DenseMatrix &vals) const;
-
-   void GetVectorValues(int i, const IntegrationRule &ir,
-                        DenseMatrix &vals, DenseMatrix &tr) const;
-
+    void GetVectorValues(int i, const IntegrationRule &ir,
+                         DenseMatrix &vals, DenseMatrix &tr) const;
+   */
    int GetFaceVectorValues(int i, int side, const IntegrationRule &ir,
                            DenseMatrix &vals, DenseMatrix &tr) const;
 
+   double GetValue(ElementTransformation &T, const IntegrationPoint &ip,
```

Comment:
Some of the methods are really just optimizations, but it's unclear that they pay off, since inside they call the more basic methods. Maybe these could be removed.

Reply:
I have rearranged and documented these methods better.

You are correct that some of the `GetValues` methods are just for convenience and not optimizations.  If the approach of using `ElementTransformation`s rather than integer indices is adopted we can look into rewriting these methods to be more efficient.